packages/server/lib/controllers/connect/postSessions.ts

```diff
@@ -0,0 +1,121 @@
+import type { PostConnectSessions } from '@nangohq/types';
+import { z } from 'zod';
+import db from '@nangohq/database';
+import { asyncWrapper } from '../../utils/asyncWrapper.js';
+import { validateRequest } from '@nangohq/utils';
+import * as keystore from '@nangohq/keystore';
+import * as linkedProfileService from '../../services/linkedProfile.service.js';
+import * as connectSessionService from '../../services/connectSession.service.js';
+
+const validate = validateRequest<PostConnectSessions>({
+    parseBody: (data) =>
+        z
+            .object({
+                linkedProfile: z.object({
```

Comment:
At this stage I would not bother allowing just sending the `profileId` because it advocate for never sending the full profile again leading to desync